src/app/frontend/common/services/global/kubeconfig.ts

```diff
@@ -0,0 +1,246 @@
+// Copyright 2017 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Inject, Injectable, NgZone} from '@angular/core';
+import {HttpErrorResponse} from '@angular/common/http';
+import {Router} from '@angular/router';
+import {CookieService} from 'ngx-cookie-service';
+import {SessionStorageService} from 'ngx-webstorage';
+import {load as loadYaml, dump as dumpYaml} from 'js-yaml';
+
+import {AsKdError, K8SError} from '../../errors/errors';
+import {AuthService} from './authentication';
+import {PluginsConfigService} from './plugin';
+import {LoginSpec} from '@api/backendapi';
+import {KdError} from '@api/frontendapi';
+import {Config, CONFIG_DI_TOKEN} from '../../../index.config';
+
+type Cluster = {
+  name: string;
+  cluster: {
+    server: string;
+    'certificate-authority-data': string;
+    'sidecar-host': string;
+  };
+};
+type User = {
+  name: string;
+  user: {
+    token: string;
+    'client-certificate-data': string;
+    'client-key-data': string;
+  };
+};
+type Context = {
+  name: string;
+  context: {
+    cluster: string;
+    user: string;
+  };
+};
+type Kubeconfig = {
+  apiVersion: string;
+  clusters: Cluster[];
+  contexts: Context[];
+  'current-context': string;
```

Comment:
There is no need to escape it. Can we change it to camelCase?

Reply:
I found that `js-yaml` can not convert keys to camelcase. Instead I found a library [`camelcase-keys`](https://github.com/sindresorhus/camelcase-keys) but it need to use `require` in `import`. Do you know better library or is it acceptable, i.e. using `require`?

Reply:
You can use require.